File: hw/cuichan/U2/HW18-Inheritance/Js/ZooApp/ec/edu/espe/zooapp/model/Mammal.js
// Mammal.js
import { Animal } from './Animal.js';

export class Mammal extends Animal {
    #hairLength;
    #mammaryGlands;

    constructor(id, name, bornOnDate, weight, sex, hairLength, mammaryGlands) {
        super(id, name, bornOnDate, weight, sex);
        this.#validateMammalParams(hairLength, mammaryGlands);
        this.#hairLength = hairLength;
        this.#mammaryGlands = mammaryGlands;
    }

    #validateMammalParams(hairLength, mammaryGlands) {
        if (hairLength < 0) throw new Error("Hair length cannot be negative");
        if (mammaryGlands < 0) throw new Error("Mammary glands count cannot be negative");
    }

    breastFeed() {
        if (this.sex !== Animal.Sex.FEMALE) {
            throw new Error("Only female mammals can breastfeed");
        }
        console.log(`Mammal ${this.name} is breastfeeding`);
    }

    get hairLength() { return this.#hairLength; }
    get mammaryGlands() { return this.#mammaryGlands; }

    toString() {
        return `Mammal {
            ${super.toString()},
            hairLength: ${this.#hairLength},
            mammaryGlands: ${this.#mammaryGlands}
        }`;
    }
}

File: hw/charij/U2/hw18Exceptions/SourceCode/ec/edu/espe/zooapp/model/Mammal.py
import Animal

class Mammal(Animal):
    def __init__(self, maximum_hair_size, mamary_glands_number, id, name, born_on_date, weight, sex):
        super().__init__(id, name, born_on_date, weight, sex)
        self.maximum_hair_size = maximum_hair_size
        self.mamary_glands_number = mamary_glands_number

    def breastfeed(self):
        print("breastfeeding the children")

    def display_info(self):
        super().display_info()
        print(f"Maximum Hair Size: {self.maximum_hair_size}")
        print(f"Mamary Glands Number: {self.mamary_glands_number}")
